### PR TITLE
GAWB-1130: New wizard for library attributes

### PR DESCRIPTION
**Pull request by MattPutnam**
Corresponding orchestration PR: https://github.com/broadinstitute/firecloud-orchestration/pull/318

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by rushtong** [comment]
Running locally, I cannot get catalog dataset attributes to save. The PUT is going through and coming back with a 200 response, but nothing seems to show up and re-editing doesn't populate the current values.

**Comment by rushtong** [comment]
There is a missing question title between `Dataset Name` and `Profiling Protocol`

screen shot 2016-12-01 at 11 52 19 am

**Comment by rushtong** [comment]
I can't really offer any feedback on the code - it's way over my head. 
I could not get this to work functionally.

**Comment by rushtong** [comment]
Ignore my previous comments - I must have had a bad branch. I can no longer reproduce the errors I saw. 

:+1 on functionality.

**Comment by rushtong** [comment]
List display bug

screen shot 2016-12-01 at 1 24 58 pm

screen shot 2016-12-01 at 1 28 06 pm

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+(defn- validate-numbers [attributes library-schema]
+  (let [numeric-props (->> attributes
+                           (utils/filter-keys #(= "integer" (get-in library-schema [:properties % :type])))
+                           (utils/map-kv (fn [k v]
+                                           [k (merge (select-keys (get-in library-schema [:properties k]) [:minimum :maximum])
+                                                     {:value v})])))
+        invalid-numbers (set (keep (fn [[k {:keys [value minimum maximum] :or {minimum -Infinity maximum Infinity}}]]
+                                     (when-not (and (= value (str (int value)))
+                                                    (<= minimum (int value) maximum))
```
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs:61]: a corner case, but this gives a false-failure on an input value of "123.0"

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+       [(if enumerate :ol :div) {}
+        (map
+          (fn [{:keys [property required inputHint renderHint]}]
+            (let [property-kwd (keyword property)
+                  {:keys [title type enum minimum consentCode]} (get-in library-schema [:properties property-kwd])
+                  error? (contains? (:invalid-properties @state) property-kwd)
+                  colorize (fn [style] (merge style (when error? {:borderColor (:exception-state style/colors)})))
+                  update-property #(swap! state update :attributes assoc property-kwd (.. % -target -value))]
+              [(if enumerate :li :div) {}
+               [:div {}
```
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: nit: consider adding `:marginBottom "2px"` (or similar) here - if the div contains an `abbr` and is followed by a text input, the underline/bottom border of the abbr is hidden.

**Comment by davidangb** [comment]
couple minor comments inline, otherwise 👍

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs:61]: I would say that's an invalid value, even though it's equivalent to an integer.  I was trying to find a good way to block invalid input but couldn't come up with anything.  Marking it as type "number" at least blocks letters and such.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs (hunk 2 above)
Comment (MattPutnam) [reply src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: Good eye!

**Pull request merged**